### PR TITLE
GitHub checks and docker auto-deploy configuration

### PR DESCRIPTION
**Pull request by andrea-deluca**
# Changes

Checks on tests and auto deploy on Docker.

## Test Check

> _These checks are performed when a pull request to the main branch is open and when a commit is pushed on main branch._

- Github Action added to perform checks on tests both in frontend and in backend. You can see the configuration file in `.github/workflows/tests.yml`.

## Docker Checks

- Github Action added to perform a check on Docker images builds. You can see the configuration file in `.github/workflows/docker-build.yml`.

   > _This action is performed when a pull request to the main branch is open._

- Github Action added to build and push images on Docker Hub. You can see the configuration file in `.github/workflows/docker-release.yml`.

   > _This action is performed when a commit is pushed on main branch and when a release is published._

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    - uses: actions/checkout@v3
+    - name: CI Backend ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+        
+    - run: npm ci
+    - run: npm run build --if-present
+    - run: npm test
```
Comment (AlbertoBaroso) [inline .github/workflows/tests.yml:61]: Add `run: npm run integration` to also execute integration tests

**Review by AlbertoBaroso — CHANGES_REQUESTED** [review]
We could also add status badges at the top of README.md such as:
`<DESCRIPTION>`

**Comment by andrea-deluca** [comment]
# Request changes

- Badge for each action added on `Readme.md` documentation file
- Command `npm run integration` added to check also backend integration tests

**Comment by andrea-deluca** [comment]
# Other changes

- Because the build is checked with docker checks, the application build is avoided for tests checks to improve computational time.

**Pull request merged**